File: backend/internal/domain/entity/page.go
package entity

import "time"

type PageStatus string

const (
	PageStatusDraft     PageStatus = "draft"
	PageStatusPublished PageStatus = "published"
)

type Page struct {
	ID          uint64
	SpaceID     uint64
	ParentID    *uint64
	Title       string
	Slug        string
	Content     string
	ContentHTML string
	Position    int
	Status      PageStatus
	Version     int
	CreatedBy   uint64
	UpdatedBy   uint64
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

func (p Page) IsRoot() bool {
	return p.ParentID == nil
}

type PageTreeNode struct {
	Page
	Children []*PageTreeNode
}

File: frontend/src/types/page.ts
export interface Page {
  id: number
  space_id: number
  parent_id: number | null
  title: string
  slug: string
  content: string
  content_html: string
  position: number
  status: 'draft' | 'published'
  version: number
  created_by: number
  updated_by: number
  created_at: string
  updated_at: string
}

export interface PageTreeNode extends Page {
  children: PageTreeNode[]
}

export interface PageVersion {
  id: number
  page_id: number
  version_number: number
  title: string
  content: string
  content_html: string
  edited_by: number
  created_at: string
}

export interface CreatePageRequest {
  space_id: number
  parent_id?: number | null
  title: string
  content?: string
}

export interface UpdatePageRequest {
  title?: string
  content?: string
  content_html?: string
  status?: string
}

export interface MovePageRequest {
  parent_id: number | null
  position: number
}
